source/common/ssl/context_impl.cc

```diff
@@ -25,7 +25,8 @@ namespace Envoy {
 namespace Ssl {
 
 ContextImpl::ContextImpl(Stats::Scope& scope, const ContextConfig& config, TimeSource& time_source)
```

Comment:
I think it would make sense to verify that you're adding at most one RSA and one ECDSA certificate, since `SSL_IDENTITY` might reject configs like that. cc @davidben

Reply:
I didn't have particular plans to reject it though, at least with the PR as-is, I don't think anything but the first ECDSA certificate and the first RSA certificate will ever be used.

Reply:
Based on what @davidben writes, I think we should be more permissive today. In general, considering that this code is throw away when SSL_IDENTITY arrives, we should aim to minimize the number of additional branches and testing complexity.

Reply:
@davidben yeah, that's my point... I'd like to prevent people from accidentally configuring two ECDSA certificates, where the second one will be always ignored, etc.

@htuch fair enough.

Reply:
@htuch I think you are confusing two related decisions.

If you only look at one curve, then obviously only one ECDSA certificate will be used and multiple are pointless. If you look at multiple curves, having both a P-256 and a P-384 certificate is potentially useful. However, P-384 and P-521 are themselves pointless, so that informs the first decision.

Reply:
I think I'm going to stick to P-256 only for simplicity and validate the certificate at config ingestion time to ensure it meets this requirement.

Reply:
See https://github.com/envoyproxy/envoy/pull/5224